metrics/options.go

```diff
@@ -79,3 +117,59 @@ func WithErrorHandler(handler errors.ErrorHandler) Option {
 		return nil
 	}
 }
+
+// WithTag sets a tag that will be sent with a specific metric.
+// Parameters:
+//   - k: The tag key
+//   - v: The tag value
+func WithTag(k, v string) MetricOptions {
+	return func(o *metricOptions) error {
+		if k == "" {
+			return fmt.Errorf("tag key cannot be empty")
+		}
+
+		if strings.ContainsAny(k, ":,|=") {
+			return fmt.Errorf("tag key contains invalid characters: %s", k)
+		}
+
+		if strings.ContainsAny(v, ":,|=") {
+			return fmt.Errorf("tag value contains invalid characters: %s", v)
+		}
+
+		if len(k)+len(v)+1 > 200 {
+			return fmt.Errorf("tag %s:%s exceeds maximum length", k, v)
+		}
+
+		for _, reservedTag := range []string{"environment", "service", "version"} {
+			if strings.ToLower(k) == reservedTag {
+				return fmt.Errorf("tag key '%s' is reserved", k)
+			}
+		}
+		o.tags = append(o.tags, fmt.Sprintf(k+":"+v))
+		return nil
+	}
+}
```

Comment:
The function and parameters for setting tags on a specific metric should match the global one in experience. Both of them should use the same validation rules.